docs/src/main/tut/xor.md

```diff
@@ -7,3 +7,338 @@ scaladoc: "#cats.data.Xor"
+    else 1.0 / i
+
+  def stringify(d: Double): String = d.toString
+}
+```
+
+Instead, let's make the fact that some of our functions can fail explicit in the return type.
+
+```tut
+object Xory {
+  def parse(s: String): Xor[NumberFormatException, Int] =
+    if (s.matches("-?[0-9]+")) Xor.right(s.toInt)
+    else Xor.left(new NumberFormatException(s"${s} is not a valid integer."))
+
+  def reciprocal(i: Int): Xor[IllegalArgumentException, Double] =
+    if (i == 0) Xor.left(new IllegalArgumentException("Cannot take reciprocal of 0."))
+    else Xor.right(1.0 / i)
+
+  def stringify(d: Double): String = d.toString
+}
+```
+
+Now, using combinators like `flatMap` and `map`, we can compose our functions together.
+
+```tut
+import Xory._
+
+def magic(s: String): Xor[Exception, String] =
+  parse(s).flatMap(reciprocal).map(stringify)
+```
+
+With the composite function that we actually care about, we can pass in strings and then pattern
+match on the exception. Because `Xor` is a sealed type (often referred to as an algebraic data type,
+or ADT), the compiler will complain if we do not check both the `Left` and `Right` case.
+
+```tut
+magic("123") match {
+  case Xor.Left(_: NumberFormatException) => println("not a number!")
+  case Xor.Left(_: IllegalArgumentException) => println("can't take reciprocal of 0!")
+  case Xor.Left(_) => println("got unknown exception")
+  case Xor.Right(s) => println(s"Got reciprocal: ${s}")
+}
+```
+
+Not bad - if we leave out any of those clauses the compiler will yell at us, as it should. However,
+note the `Xor.Left(_)` clause - the compiler will complain if we leave that out because it knows
+that given the type `Xor[Exception, String]`, there can be inhabitants of `Xor.Left` that are not
+`NumberFormatException` or `IllegalArgumentException`. However, we "know" by inspection of the source
+that those will be the only exceptions thrown, so it seems strange to have to account for other exceptions.
+This implies that our choice of type is not strong enough.
+
+### Example usage: Round 2
+Instead of using exceptions as our error value, let's instead enumerate explicitly the things that
+can go wrong in our program.
+
+```tut
+object Xory2 {
+  sealed abstract class Error
+  final case class NotANumber(string: String) extends Error
+  final case object NoZeroReciprocal extends Error
+
+  def parse(s: String): Xor[Error, Int] =
+    if (s.matches("-?[0-9]+")) Xor.right(s.toInt)
+    else Xor.left(NotANumber(s))
+
+  def reciprocal(i: Int): Xor[Error, Double] =
+    if (i == 0) Xor.left(NoZeroReciprocal)
+    else Xor.right(1.0 / i)
+
+  def stringify(d: Double): String = d.toString
+
+  def magic(s: String): Xor[Error, String] =
+    parse(s).flatMap(reciprocal).map(stringify)
+}
+```
+
+For our little module, we enumerate any and all errors that can occur. Then, instead of using
+exception classes as error values, we use one of the enumerated cases. Now when we pattern
+match, we get much nicer matching.
```

Comment:
Maybe just another quick aside that since `Error` is sealed, no third-party can add additional subtypes which we might fail to handle. You mentioned sealed ADTs earlier, so maybe this is redundant, but it seems like an important point.